library/src/main/kotlin/org/quiltmc/qkl/library/networking/serialization/serializedPacketRegistration.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.library.networking.serialization
+
+import kotlinx.serialization.modules.EmptySerializersModule
+import kotlinx.serialization.modules.SerializersModule
+import net.minecraft.client.MinecraftClient
+import net.minecraft.client.network.ClientPlayNetworkHandler
+import net.minecraft.server.MinecraftServer
+import net.minecraft.server.network.ServerPlayNetworkHandler
+import net.minecraft.server.network.ServerPlayerEntity
+import net.minecraft.util.Identifier
+import org.quiltmc.qsl.networking.api.PacketSender
+import org.quiltmc.qsl.networking.api.ServerPlayNetworking
+import org.quiltmc.qsl.networking.api.client.ClientPlayNetworking
+
+public typealias OnSerializedPacketClient<T> =
+        T.(MinecraftClient, ClientPlayNetworkHandler, PacketSender)->Unit
+public typealias OnSerializedPacketServer<T> =
+        T.(MinecraftServer, ServerPlayerEntity, ServerPlayNetworkHandler, PacketSender)->Unit
+
+/**
+ * A utility class for serialized packet registration.
+ * @param id The packet channel to expect this packet on.
+ * @param direction The directions that this packet can be sent in.
+ * @param serializersModule The serializers module to be used for decoding
+ */
+public class SerializedPacketRegistration<P>(
```

Comment:
It may be reasonable to add an alternative to this supporting class-based declarations in addition to the DSL-style declaration.

Reply:
Given theres no namespaces yet. ill probably throw that on the other objects i created